app/views/partials/_route_planning.erb

```diff
@@ -0,0 +1,36 @@
+
+<h3>Route Planning</h3>
+
+<div id="map-canvas"></div>
+
+
+
+
+
+
+  <script>
+    //this can be refactored into the head of the partial by using "named yeilds" in the layout and "content for" in the partial... probably should do that
+
+    // turns out this isn't supported by basic sinatra, although it is built in functionality in rails
+    function initialize() {
```

Comment:
your javascript should live in a javascript file, not just a script tag, then you should link to it in your head